### PR TITLE
fix: alias api

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

I've fixed the declarations and implementation of the `alias` API.
- Marked `alias(previousId, newId)` as deprecated.
- Added the correct function declaration `alias(newId, [options])` to the API.
- The API was also fixed to return the promise from the bridge.
  - We have to do this for all the other APIs as well later.

Misc:
- Updated the sample app with all the correct usages of the `alias` API.
- Also, added the missing `package-lock.json` file.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: apps/example/src/app/RudderEvents.tsx [hunk 1]
```diff
@@ -101,7 +101,16 @@ const RudderEvents = () => {
...
   const alias = () => {
-    rudderClient.alias('RN Alias ID', 'previous RN userId');
+    rudderClient.alias('New User ID');
+
+    // All the below calls are also valid usages of alias
+    // Anything else is invalid
+    // rudderClient.alias('New User ID', null);
+    // rudderClient.alias('New User ID', {});
+    // rudderClient.alias('New User ID', undefined);
+
```
Comment (1abhishekpandey) [inline apps/example/src/app/RudderEvents.tsx:111]: Let's add the last case which highlights the steps to pass the RudderOptions object.
```suggestion
    // rudderClient.alias("New User ID", getLocalOptions());

```

**Review by 1abhishekpandey — COMMENTED** [review]
Very nice!

File: libs/sdk/src/RudderClient.ts [hunk 2]
```diff
@@ -209,22 +209,32 @@ async function group(
...
-    bridge.alias(newId, filterNaN(options));
+    return Promise.resolve();
+  }
+  // this is to support the old alias method
+  // alias('previousId', 'newId')
+  // The previous ID is ignored.
+  if (typeof newIdOrOptions == 'string') {
+    return bridge.alias(newIdOrOptions, null);
+  } else if (typeof newIdOrOptions == 'object' && !Array.isArray(newIdOrOptions)) {
+    return bridge.alias(newOrPrevId, newIdOrOptions);
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply libs/sdk/src/RudderClient.ts]: Done.

File: apps/example/src/app/RudderEvents.tsx (hunk 1 above)
Comment (saikumarrs) [reply apps/example/src/app/RudderEvents.tsx:111]: Done.

**Review by 1abhishekpandey — APPROVED** [review]

**Pull request merged**